ksql-engine/src/main/java/io/confluent/ksql/util/ExpressionTypeManager.java

```diff
@@ -265,11 +271,28 @@ protected Expression visitFunctionCall(
     return null;
   }
 
-  private static Schema resolveArithmeticType(final Schema leftSchema,
-      final Schema rightSchema) {
+  private static Schema resolveArithmeticType(final Schema leftSchema, final Schema rightSchema) {
     return SchemaUtil.resolveArithmeticType(leftSchema.type(), rightSchema.type());
   }
 
+  private static void isValidComparison(
+      final Schema.Type leftType,
+      final ComparisonExpression.Type operator,
+      final Schema.Type rightType) {
+    if (ComparisonUtil.areCompatibleTypesForComparison(leftType, rightType)) {
```

Comment:
Would be easier to grok if you invert the first if and combine the two nested ifs.

```java
    if (!ComparisonUtil.areCompatibleTypesForComparison(leftType, rightType)) {
       throw new KsqlException("Operator " + operator + " cannot be used to compare " + leftTyp  + " and " + rightType);
   }
   
   if (leftType == Schema.Type.BOOLEAN && operator != Type.EQUAL && operator != Type.NOT_EQUAL) {
      throw new KsqlException("Operator " + operator + " cannot be used to compare " + leftType  + " and " + rightType);
   }
```

Reply:
Moved this to 'ComparisonUtil` and simplified the ifs as you suggested.